python-squid/src/main/java/org/sonar/python/semantic/SymbolTableBuilderVisitor.java

```diff
@@ -305,13 +391,15 @@ private void addNonlocalName(String name) {
   private static class SymbolImpl implements Symbol {
 
     private final String name;
+    private String moduleName;
```

Comment:
I think it would be cleaner to have this field initialized in constructor and make it final. That way the contract is clear, that it cannot be modified once symbol is created